### ISSUE TITLE
Is SoftmaxCrossEntropyLoss with sparseLabels (multi-class) working correctly?

### ISSUE THREAD

**Issue by ux51ekac** [deepjavalibrary/djl#2107]
For a token classification task I use the SoftmaxCrossEntropyLoss. The predictions have shape `(16, 512, 2)`, the correct labels have shape `(16, 512, 1)`, given that the batch size is 16 and I have at most 512 tokens per document.

Since I wanted to weight my imbalanced classes, I debugged through the SCELoss code and the behavior seemed strange to me:
When calculating the loss in [`evaluate`](https://github.com/deepjavalibrary/djl/blob/b7fce8ca967a51988ee36f5dde368f525237c5c8/api/src/main/java/ai/djl/training/loss/SoftmaxCrossEntropyLoss.java#L73) the predictions that are picked with the calculated [`pickIndex`](https://github.com/deepjavalibrary/djl/blob/b7fce8ca967a51988ee36f5dde368f525237c5c8/api/src/main/java/ai/djl/training/loss/SoftmaxCrossEntropyLoss.java#L85) is `(16, 512, 1)` which (if I understand correctly) just picks the part of the first class of the predictions into account for the loss value. I am not sure if that is true or this behavior is intended, but if so, it might explain why my model optimizes to only use the first class.

**Comment by ux51ekac** [comment]
> First of all, there is a flexibility of building your own lossfunction class, like what is shown in this transfer learning feature:

Yes I am doing exactly that.

> Make sure you are choosing the right option of this parameter

Is also what I am doing. `sparseLabel` is true in my case and that is what it should be.

> is not the expected behaviour. To further debug, could you provide a sample array at that step and show your expected behaviour?

How exactly do you want me to provide this information?
